sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -96,19 +95,13 @@ object AnalysisContext {
 class Analyzer(
     catalog: SessionCatalog,
     conf: SQLConf,
-    maxIterations: Int,
-    override val lookupCatalog: Option[(String) => CatalogPlugin] = None)
-  extends RuleExecutor[LogicalPlan] with CheckAnalysis with LookupCatalog {
+    maxIterations: Int)
+  extends RuleExecutor[LogicalPlan] with CheckAnalysis {
```

Comment:
I use this for `FunctionCatalog` in the rules that are defined as objects in the `Analyzer`. I'd rather keep it in. At a minimum, this change should be in a separate PR.

Reply:
Will keep it.